### PR TITLE
Adds SSL connector configuration (HTTP2) to Tomcat

### PR DESCRIPTION
**Pull request by mmadzin**
Adds SSL connector configuration (HTTP2) to Tomcat

### CODE REVIEW

File: core/src/main/groovy/noe/server/Tomcat.groovy [hunk 1]
```diff
@@ -669,6 +669,28 @@ class Tomcat extends ServerAbstract implements WorkerServer {
...
   }
 
+  void enableSslNIOProtocolOpenSsl(String protocol = "org.apache.coyote.http11.Http11NioProtocol",
+                                   boolean upgradeProtocol = false) {
+    def nl = platform.nl
+    def dummyComment = '<!-- Define an AJP 1.3 Connector on port 8009 -->'
+
+    def enableNIOSsl = '<Connector port="' + this.mainHttpsPort.toString() + '" protocol="' + protocol + '"' + nl +
+      'sslImplementationName="org.apache.tomcat.util.net.openssl.OpenSSLImplementation"' + nl +
+      'maxThreads="150" SSLEnabled="true">' + nl + nl
```
Comment (jan-onderka) [inline core/src/main/groovy/noe/server/Tomcat.groovy]: Make `maxThreads` also configurable

**Comment by jan-onderka** [comment]
There might be better solution for it via class `SecureHttpConnectorTomcat`

**Comment by jstefl** [comment]
This kind of implementation was so-so OK in times we did manipulate with configurations files by binding on some comments and did some replacements. There was no common XML tools in play.

Now, unfortunately, this is not reliable, because we use XML tools(for instance in `TomcatConfigurator`) and they removes all comets from code, so this method can stop working mysteriously not to mention that executor is not informed whether config change was successful or not.

I understand, there is code, which has described flaw, however lets not spread this problem if we have relevant tool to our disposal.

I recommend 
 * Enhance `TomcatConfigurator` ( resp. `SecureHttpConnectorTomcat`, `ConnectorAttributesTransformer` ...) and use `TomcatConfigurator` directly (to not create circular dependency by using it method in `Tomcat`).

**Comment by Karm** [comment]
@mmadzin @jstefl @jan-onderka 
The implementation is wrong - it uses old approach. Look up the JSSE and JSSE OpenSSL and Http2Upgrade XMLSlurper configs in Noe-Tests -> modcluster module and feel free to opensource them :smiley:

**Pull request closed without merging**